File: TMessagesProj/src/main/java/tw/nekomimi/nekogram/utils/UIUtil.java
package tw.nekomimi.nekogram.utils;
import android.os.Handler;
import android.os.Looper;
public class UIUtil {
    private static final Handler handler = new Handler(Looper.getMainLooper());
    public static void runOnUIThread(Runnable r) { handler.post(r); }
    public static void runOnIoDispatcher(Runnable r) { new Thread(r).start(); }
    public static void post(Runnable r) { handler.post(r); }
}

File: TMessagesProj/src/main/java/tw/nekomimi/nekogram/utils/UIUtil.kt
package tw.nekomimi.nekogram.utils

object UIUtil {

    @JvmStatic
    fun runOnUIThread(runnable: Runnable) {
        runnable.run()
    }

    fun runOnUIThread(runnable: () -> Unit) {
        runnable()
    }

    @JvmStatic
    @JvmOverloads
    fun runOnIoDispatcher(runnable: Runnable, delay: Long = 0) {
        runnable.run()
    }

    fun runOnIoDispatcher(runnable: suspend () -> Unit) {}
}
